lib/node_modules/@stdlib/time/base/parse-duration/lib/main.js

```diff
@@ -0,0 +1,79 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// VARIABLES //
+
+var RE_TIME_UNIT = /(\d+)([a-z]+)/gi;
+
+
+// MAIN //
+
+/**
+* Parses a duration string such as `1m3s10ms` into an object with `days`, `hours`, `minutes`, `seconds`, and `milliseconds`.
+*
+* @param {string} str - duration string
+* @returns {Object} duration object
+*
+* @example
+* var obj = parseDuration( '1m3s10ms' );
+* // returns { 'days': 0, 'hours': 0, 'minutes': 1, 'seconds': 3, 'milliseconds': 10 }
+*
+* @example
+* var obj = parseDuration( '1m3s' );
+* // returns { 'days': 0, 'hours': 0, 'minutes': 1, 'seconds': 3, 'milliseconds': 0 }
+*/
+function parseDuration( str ) {
+	var parts;
+	var out;
+	var i;
+
+	out = {
+		'days': 0,
+		'hours': 0,
+		'minutes': 0,
+		'seconds': 0,
+		'milliseconds': 0
+	};
+	if ( str.length === 0 ) {
+		return out;
+	}
+	parts = RE_TIME_UNIT.exec( str );
+	while ( parts ) {
+		i = parts[ 2 ];
+		if ( i === 'd' || i === 'D' ) {
```

Comment:
Instead of the more complicated if/else logic, why not do `RE_TIME_UNIT.exec( lowercase( str ) )`?

Otherwise, for L68, we also need to accommodate `Ms` and `mS`.

Reply:
Changed.